### PR TITLE
core/vm: Add Venoki header's fields when verify double sign

### PR DESCRIPTION
**Pull request by chiphamskymavis**
After Venoki, these fields are added into block's header: BaseFee, BlodGasUsed, ExceedBlobGas.
We need to add them into the double sign verification process to make sure the logic up to date.

The separated struct due to lack off optional and null support of contract Abi

### CODE REVIEW

File: core/types/block.go [hunk 1]
```diff
@@ -98,20 +109,68 @@ func FromHeader(header *Header, chainId *big.Int) *BlockHeader {
...
+		BaseFee:     header.BaseFee,
 	}
 	copy(blockHeader.ParentHash[:], header.ParentHash.Bytes())
 	copy(blockHeader.StateRoot[:], header.Root.Bytes())
 	copy(blockHeader.TransactionsRoot[:], header.TxHash.Bytes())
 	copy(blockHeader.ReceiptsRoot[:], header.ReceiptHash.Bytes())
 	copy(blockHeader.MixHash[:], header.MixDigest.Bytes())
+	if header.BaseFee == nil {
+		blockHeader.BaseFee = nil
+	}
```
Comment (minh-bq) [inline core/types/block.go:121]: This is redundant, can be removed

File: core/vm/consortium_precompiled_contracts.go [hunk 2]
```diff
@@ -516,20 +543,22 @@ func (c *consortiumVerifyHeaders) getSigner(header types.BlockHeader) (common.Ad
 	return signer, nil
 }
 
+// verify if two has same signers, same block, same consensus address, then it is double signed
+// return false if it is not double signed
```
Comment (minh-bq) [inline core/vm/consortium_precompiled_contracts.go:547]:
```suggestion
// verify returns true if 2 blocks has the same signer (consensus address), same block number
// but with different seal hash
```

**Review by minh-bq — COMMENTED** [review]
I've just left some small comments. Overall, LGTM.

File: core/vm/consortium_precompiled_contracts.go (hunk 2 above)
Comment (chiphamskymavis) [reply core/vm/consortium_precompiled_contracts.go]: Thank you, updated

File: core/types/block.go (hunk 1 above)
Comment (chiphamskymavis) [reply core/types/block.go]: Oh, nice catch, updated this.

**Review by minh-bq — APPROVED** [review]

**Pull request merged**